internal/p2p/peermanager.go

```diff
@@ -1254,6 +1333,43 @@ func (p *peerInfo) Copy() peerInfo {
 	return c
 }
 
+// LastDialed returns when the peer was last dialed, and if that dial
+// attempt was successful.
+func (p *peerInfo) LastDialed() (time.Time, bool) {
+	var (
+		last    time.Time
+		success bool
+	)
+	last = last.Add(-1) // so it's after the epoch
+
+	for _, addr := range p.AddressInfo {
+		if addr.LastDialFailure.Equal(addr.LastDialSuccess) {
+			if addr.LastDialFailure.IsZero() {
+				continue
+			}
+			if last.Before(addr.LastDialSuccess) {
+				last = addr.LastDialSuccess
+				success = true
+			}
+		}
+		if addr.LastDialFailure.Before(addr.LastDialSuccess) {
+			if last.Before(addr.LastDialFailure) {
+				continue
+			}
+			last = addr.LastDialFailure
+			success = false
+		}
+	}
+
+	// if we never modified last, then we should return it to the
+	// zero value
+	if last.Add(1).IsZero() {
+		last = last.Add(1)
```

Comment:
```suggestion
		return time.Time{}, success
```

since we really do just want zero and being clear that this is what we got is easier if we just return it.